### PR TITLE
Switch to Pytest

### PR DESCRIPTION
**Pull request by ArtemGordinsky**
Mostly just to get `parametrize` for free. Also, `pytest` is just better.

### CODE REVIEW

File: freshdesk/v2/test.py [hunk 1]
```diff
@@ -110,73 +116,31 @@ def _delete(self, url, *args, **kwargs):
         raise HTTPError('404: mocked_api_delete() has no pattern for \'{}\''.format(url))    
 
 
-class TestAPIClass(TestCase):
+class TestAPIClass:
```
Comment (ArtemGordinsky) [inline freshdesk/v2/test.py]: Pytest's `parametrize` doesn't work with `unittest` test cases, so I had to drop it here.

Comment (sjkingo) [reply freshdesk/v2/test.py]: Fine with me, I don't use any base classes with pytest either. Can you remove the unused `unittest` import though?

File: freshdesk/v1/test.py [hunk 2]
```diff
@@ -254,11 +254,11 @@ def test_get_contact(self):
     def test_list_contacts(self):
         contacts = self.api.contacts.list_contacts()
         self.assertIsInstance(contacts, list)
```
Comment (sjkingo) [inline freshdesk/v1/test.py]: Maybe a bit pedantic, but do you want to change these `assertEqual`'s to plain `assert`'s for consistency?

Comment (ArtemGordinsky) [reply freshdesk/v1/test.py]: Sure, why not.

File: freshdesk/v2/test.py (hunk 1 above)
Comment (ArtemGordinsky) [reply freshdesk/v2/test.py]: `unittest` was still used by every other test class but I refactored them as well now to be consistent.

**Comment by ArtemGordinsky** [comment]
> +1,109 −987

Not really a "drop-in replacement" anymore, sorry 😅 

Let me know what do you think about the new structure of the tests. Also, feel free to revert the last commit if you think that's too much.

**Comment by sjkingo** [comment]
Looks awesome to me - merged. Cheers

**Comment by ArtemGordinsky** [comment]
Thanks! Could you do a release on pypi as well? Would be nice to start using the new error classes in my project.

**Comment by sjkingo** [comment]
Done! Version 1.2.5 should be up now. Thanks for the help!

**Pull request merged**